core/zoe/contracts/simpleOffer/srcs/secondPriceSrcs.js

```diff
@@ -0,0 +1,107 @@
+import harden from '@agoric/harden';
+
+import { makeHasOkRules } from '../../../contractUtils';
+
+const hasOkRulesInitialOffer = makeHasOkRules([
+  ['haveExactly', 'wantAtLeast'], // The initial offer
+]);
+
+const hasOkRulesBids = makeHasOkRules([
+  ['wantExactly', 'haveAtMost'], // Subsequent bids
+]);
+
+const isValidInitialOfferDesc = newOfferDesc =>
+  hasOkRulesInitialOffer(newOfferDesc);
+
+const isValidBid = (assays, initialOffer, newBid) =>
+  hasOkRulesBids(newBid) &&
+  assays[1].includes(newBid[1].amount, initialOffer[1].amount);
+
+const isValidOffer = (
+  assays,
+  offerIds,
+  offerIdsToOfferDescs,
+  offerMadeDesc,
+) => {
+  const isInitialOffer = offerIds.length === 0;
+  return (
+    (isInitialOffer && isValidInitialOfferDesc(offerMadeDesc)) ||
+    (!isInitialOffer &&
+      isValidBid(assays, offerIdsToOfferDescs.get(offerIds[0]), offerMadeDesc))
+  );
+};
+
+// Iterate over all of the bids keeping the highest and second highest bid.
+const findWinnerAndPrice = (strategy, offerIds, bids) => {
```

Comment:
This code assumes that the `initialOffer` isn't included in `offerIds`. Would it make sense to rename it `bidIds`?

Reply:
Yup